pkg/destroy/aws/aws.go

```diff
@@ -1716,6 +1716,10 @@ func deleteRoute53(session *session.Session, arn arn.ARN, logger logrus.FieldLog
 
 	sharedZoneID, err := getSharedHostedZone(client, id, logger)
 	if err != nil {
+		if err.(awserr.Error).Code() == route53.ErrCodeNoSuchHostedZone {
+			logger.Warn("hosted zone in list of tagged resources but no longer exists, skipping route53 cleanup")
+			return nil
```

Comment:
Should we return nil? I would think that if there is no sharedZone then the resources that are deleted after this would also not exist, but here: https://github.com/openshift/installer/blob/5df5607169bb8b8c878b8b5cc108d1a6d0fec0d7/pkg/destroy/aws/aws.go#L1733

we proceed even if no sharedZone exists...

Reply:
The not-found error here is for the case where the private zone does not exist. If the private zone exists, but there is no public zone, then `getSharedHostedZone` will return an empty string and a nil error.

Reply:
Thanks for clarifying.

As mentioned, this seems like an AWS problem so it is beyond our control to fix. 

/lgtm
/approve

Reply:
I am really unsure, sharedZoneID is used again on line 1761 so that's going to probably fail, line 1770 maybe? I don't know this code or route53 very well, it feels like it's game over if we can't get the shared zone ID.

Reply:
Oops was looking at an old load of the tab. :)